storm/src/test/storm/mesos/MesosNimbusTest.java

```diff
@@ -17,304 +17,451 @@
-            3.0,
-            "mem"
-        )
-    );
-
-    assertEquals(
-        Arrays.asList(
-            TestUtils.buildScalarResource("cpus", 1.0),
-            TestUtils.buildScalarResourceWithRole("cpus", 3.0, "reserved")
-        ),
-        mesosNimbus.getResourcesScalar(
-            TestUtils.buildResourceList(1, 2, 3, 4),
-            4.0,
-            "cpus"
-        )
-    );
-
-    assertEquals(
-        Arrays.asList(
-            TestUtils.buildScalarResource("mem", 2.0),
-            TestUtils.buildScalarResourceWithRole("mem", 4.0, "reserved")
-        ),
-        mesosNimbus.getResourcesScalar(
-            TestUtils.buildResourceList(1, 2, 3, 4),
-            6.0,
-            "mem"
-        )
-    );
-
-    assertEquals(
-        Arrays.asList(
-            TestUtils.buildScalarResource("cpus", 1.0),
-            TestUtils.buildScalarResourceWithRole("cpus", 0.5, "reserved")
-        ),
-        mesosNimbus.getResourcesScalar(
-            TestUtils.buildResourceList(1, 2, 3, 4),
-            1.5,
-            "cpus"
-        )
-    );
-
-    assertEquals(
-        Arrays.asList(
-            TestUtils.buildScalarResource("mem", 2.0),
-            TestUtils.buildScalarResourceWithRole("mem", 0.5, "reserved")
-        ),
-        mesosNimbus.getResourcesScalar(
-            TestUtils.buildResourceList(1, 2, 3, 4),
-            2.5,
-            "mem"
-        )
-    );
+  public static final Integer DEFAULT_WORKER_COUNT = 2;
+  public static final Integer DEFAULT_BUCKET_COUNT = 2;
+  public static final String FRAMEWORK_ROLE = "staas";
+
+  private RotatingMap<OfferID, Offer> rotatingMap = null;
+  Map<String, Collection<WorkerSlot>> slotsForTopologiesNeedingAssignments = null;
+  MesosNimbus mesosNimbus = null;
+  Map<String, String> mesosStormConf;
+
+  private boolean hasResources(String role, List<Protos.Resource> resourceList, Double cpus, Double mem, Long port) {
+    Double actualCpu = 0.0d, actualMem = 0.0d;
+    List<Long> expectedPorts = new ArrayList<>();
+    expectedPorts.add(port);
+
+    List<Long> actualPorts = new ArrayList<>();
+    for (Protos.Resource resource : resourceList) {
+      if (!resource.getRole().equals(role)) {
+        continue;
+      }
+      ResourceType r = ResourceType.of(resource.getName());
+      switch (r) {
+        case CPU:
+          actualCpu += resource.getScalar().getValue();
+          break;
+        case MEM:
+          actualMem += resource.getScalar().getValue();
+          break;
+        case PORTS:
+          Protos.Value.Ranges ranges = resource.getRanges();
+          for (Protos.Value.Range range : ranges.getRangeList()) {
+            for (long i = 0; i < (range.getEnd() - range.getBegin() + 1); i++) {
+              actualPorts.add(range.getBegin() + i);
+            }
+          }
+      }
+    }
+
+    boolean hasPorts = (expectedPorts.size() == actualPorts.size()) && expectedPorts.containsAll(actualPorts);
```

Comment:
If both `expectedPorts` and `actualPorts` have nothing in them, then this will be true, even if we don't have any ports. If that's the expected behavior, this variable name should be changed to something like `allExpectedPortsFound`.